src/export/InstanceExporter.ts

```diff
@@ -434,6 +434,27 @@ export class InstanceExporter implements Fishable {
             break;
           }
         }
+        // if the choice only has one type, we may not have created the choice slice.
+        // if this happens, everything is fine.
+        if (instanceChild == null) {
+          const singleTypeChoiceElements = possibleChoiceSlices.filter(
+            c => c.path === child.path && c.type.length === 1
+          );
+          for (const choiceElement of singleTypeChoiceElements) {
+            const sliceName = childPathEnd.replace('[x]', upperFirst(choiceElement.type[0].code));
+            instanceChild = instance[`_${sliceName}`] ?? instance[sliceName];
+            const splitPath = splitOnPathPeriods(choiceElement.id);
+            if (
+              instanceChild != null &&
+              (instance._sliceName
+                ? splitPath[splitPath.length - 2].split(':')[1] === instance._sliceName
+                : true)
+            ) {
+              child = choiceElement;
+              break;
+            }
+          }
+        }
```

Comment:
This seems to work well, but it seems pretty similar to the block above it (on lines 414 - 436). The two main differences seem to be the condition in the choiceSlices filter and the slice name used to get the instanceChild. I briefly tried to update the [filter on line 419](https://github.com/FHIR/sushi/pull/1446/files#diff-7ccabd62381ce8ebf05457c24cc3884eeefb1ec67ff259b82ca7f2c14c4f51cfR419) to be something like this:

```
const choiceSlices = possibleChoiceSlices.filter(
  c => c.path === child.path && (c.sliceName || c.type.length === 1)
);
```
And then I made a variable to either use the `choiceSlice.sliceName` or replace the `[x]` portion like you added and used that variable when setting the `instanceChild` [on line 422](https://github.com/FHIR/sushi/pull/1446/files#diff-7ccabd62381ce8ebf05457c24cc3884eeefb1ec67ff259b82ca7f2c14c4f51cfR422), and things seemed to be okay.

This seems like it might reduce some duplicate code and reduce the chance that we might edit one of these without editing the other. But I'm curious what your thoughts are. I also didn't thoroughly test this, it was more just a quick attempt on my end, so there might be a reason that I missed that these two cases are different.

Reply:
Ohhh... smart. I like it!